Composed/Core/Deprecations.swift

```diff
@@ -0,0 +1,90 @@
+import Foundation
+
+public extension DataStoreDelegate {
+    @available(swift, obsoleted: 1.0, message: "This is no longer required and has been removed entirely. Calling this method results in a fatalError")
```

Comment:
I think this is saying it was obsoleted in swift 1.0. Probably won't change anything but I think this should be `@available(*, obsoleted, message "...")`

Reply:
I know, I would love to do that, but its not supported ;) 
This is the only way I could do it, or deprecated, but I preferred an error on usage since these are just for migration at this stage, they're not likely to stay very long.

Reply:
27f88d2805a12916bffa58c809fbaf104832bc83